### ISSUE TITLE
Where is the ObjectiveC part?

### ISSUE THREAD

**Issue by vgorloff** [serdarbakirtas/ToDo#1]
The requirements below still missed:

```

46. When creating Framework target choose the following options:”Language” – 
Objective-C, “Include Tests” – checks is on.

47. After Framework target is created, go to “Build setting”, find an option 
“Objective-C Automatic Reference Counting”, and switch it for the 
Framework target to “No”. 

```

**Comment by serdarbakirtas** [comment]
> The major requirements below still missed:
> 
> ```
> 
> 46. When creating Framework target choose the following options:”Language” – 
> Objective-C, “Include Tests” – checks is on.
> 
> 47. After Framework target is created, go to “Build setting”, find an option 
> “Objective-C Automatic Reference Counting”, and switch it for the 
> Framework target to “No”. 
> ```
> 
> This Code Challenge without ObjC part (with ARC feature switched OFF) is not a Code Challenge ;)

Dear Garlov,

Please collaborate your HR department, I already contact with them about these topics.

My message: Some comments about the tech side: 
When I checked the details of the technical task, I realized that there are some Objective-C skills I need to use, and since I don't have much experience with it I decided to keep writing this part in Swift, I would like you to discuss this with your team before the interview.

It is necessary to discuss what percentage of your project currently covers Objective-C if a significant part of it is written in Objective-C language. Unfortunately, I am lacking in this regard. I had a lot of difficulty remembering the language I used years ago.

If we can continue in this way, they can check the assignment I have attached and we can discuss the details tomorrow. But again, if objective-c is an essential part of the project, I suggest canceling tomorrow's meeting.


**Your HR team's message:** Yes, we can continue the interviewing process. Our project doesn't have/require so much Objective C knowledge.

**Comment by vgorloff** [comment]
> But I set the “Objective-C Automatic Reference Counting” for the framework target as a NO.

Even if you set “Objective-C Automatic Reference Counting” for the framework target as a NO, there is still no ObjC code inside. The following requirement is missed:

```
48. Implement Application Data model inside a Framework target using Objective-C 
programming language with manual memory management, by sending retain, 
release, or autorelease messages to objects. 
```

**Comment by serdarbakirtas** [comment]
> > But I set the “Objective-C Automatic Reference Counting” for the framework target as a NO.
> 
> Even if you set “Objective-C Automatic Reference Counting” for the framework target as a NO, there is still no ObjC code inside. The following requirement is missed:
> 
> ```
> 48. Implement Application Data model inside a Framework target using Objective-C 
> programming language with manual memory management, by sending retain, 
> release, or autorelease messages to objects. 
> ```

Yes, I know. But as I mentioned in my previous comment _**I do not have since I don't have much experience with objective-C I decided to keep writing this part in Swift**_ 

Please consider it. If your company’s project requires so much Objective C knowledge. We should cancel our progress.

But, your HR team said to me that  **Our project doesn't have/require so much Objective C knowledge.**

Unfortunately, I can not change the Data modal to the Objective-C.

**Comment by vgorloff** [comment]
Don't want to continue discussion, but let me share two facts:

1. When I had an interview on Sr. iOS developer position, one company sent me task on C# language. I made it and won the position. Was not using C# for ~ 8 years. Why HR sent me C# task ??? still a big question ;) 

2. One candidate during hiring process said that he is not strong enough in ObjC, and converted Swift to ObjC by using AI ;)

**Comment by serdarbakirtas** [comment]
> Don't want to continue discussion, but let me share two facts:
> 
> 1. When I had an interview on Sr. iOS developer position, one company sent me task on C# language. I made it and won the position. Was not using C# for ~ 8 years. Why HR sent me C# task ??? still a big question ;)
> 2. One candidate during hiring process said that he is not strong enough in ObjC, and converted Swift to ObjC by using AI ;)

Of course, I can also use ChatGPT to migrate Swift to Objective-C. But that doesn't make me a senior.
I wanted to be very transparent about this.

**Comment by vgorloff** [comment]
Not sure that ChatGPT will generate Objective-C code which will not crash and will not have memory leaks. Probably some manual maintenance will be needed anyway. But, it just an example that it is a doable task.

–––
Wish you a nice day,
Vlad

**Comment by serdarbakirtas** [comment]
> Not sure that ChatGPT will generate Objective-C code which will not crash and will not have memory leaks. Probably some manual maintenance will be needed anyway. But, it just an example that it is a doable task.
> 
> ––– Wish you a nice day, Vlad

You are completely right, We can find a solution for this part somehow. This is a doable task as you mentioned.

Actually, I just wanted to say that if I'm going to work on Objective-C-heavy work in an internal project in this Company, it might be challenging and troublesome for me.

I appreciate your comments. I am ready to collaborate with you on any tasks/issues :) 

---
Have a nice day
Serdar
